sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts

```diff
@@ -281,27 +306,109 @@ export class ECertGenerationService {
+            (disbursementValue.disbursedAmountSubtracted ?? 0) -
+            (disbursementValue.overawardAmountSubtracted ?? 0);
           disbursementValue.effectiveAmount = 0;
+          disbursementValue.restrictionSubtracted = {
+            id: disbursement.restrictionId,
+          } as Restriction;
         } else {
           const effectiveValue =
             disbursementValue.valueAmount -
             (disbursementValue.disbursedAmountSubtracted ?? 0) -
             (disbursementValue.overawardAmountSubtracted ?? 0);
           disbursementValue.effectiveAmount = round(effectiveValue);
+
+          const application = disbursement.studentAssessment.application;
+          await this.checkLifeTimeMaximumAndAddStudentRestriction(
+            disbursementValue,
+            application,
+            entityManager,
+          );
         }
       }
     }
   }
 
+  /**
+   * Check if the student hits/exceeds the life time maximum for the BCSL Award.
+   * If they hits/exceeds the life time maximum, then the award is reduced so the
+   * student hits the exact maximum value and the student restriction
+   * {@link  RestrictionCode.BCLM} is placed for the student.
+   * @param disbursementValue disbursement value.
+   * @param application application related to the disbursement.
+   * @param entityManager used to execute the commands in the same transaction.
+   */
+  private async checkLifeTimeMaximumAndAddStudentRestriction(
+    disbursementValue: DisbursementValue,
+    application: Application,
+    entityManager: EntityManager,
+  ): Promise<void> {
+    if (disbursementValue.valueType === DisbursementValueType.BCLoan) {
+      const student = application.student;
+      const maxLifetimeBCLoanAmount =
+        application.programYear.maxLifetimeBCLoanAmount;
+      const auditUser = await this.systemUsersService.systemUser();
+
+      const totalLifeTimeAmount =
+        (await this.sfasApplicationService.totalLegacyBCSLAmount(
+          student.sinValidation.sin,
+          student.birthDate,
+          student.user.lastName,
+        )) +
+        (await this.disbursementScheduleSharedService.totalDisbursedBCSLAmount(
+          student.id,
+        )) +
+        disbursementValue.effectiveAmount;
+
+      if (totalLifeTimeAmount >= maxLifetimeBCLoanAmount) {
+        // Amount subtracted when lifetime maximum is reached.
+        const amountSubtracted = totalLifeTimeAmount - maxLifetimeBCLoanAmount;
+        // Ideally disbursementValue.effectiveAmount should be greater or equal to amountSubtracted.
+        // The flow will not reach here if the ministry ignore the restriction for the previous
+        // disbursement/application and money went out to the student, even though they reach the maximum.
+        const newEffectiveAmount =
+          disbursementValue.effectiveAmount - amountSubtracted;
+        /**
+         * Create {@link RestrictionCode.BCLM} restriction when lifetime maximum is reached/exceeded.
```

Comment:
I like the `Create {@link RestrictionCode.BCLM}` and I would support its usage in the methods comments.
At this level, I believe that it leaves the code less readable, also because you need the comment to be placed with `/***/`.